src/components/tabs/tab.component.ts

```diff
@@ -1,14 +1,23 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, TemplateRef, ContentChild, ElementRef } from '@angular/core';
+import { IfTabActiveDirective } from './if-tab-active.directive';
 
 /**
  * TODO: Remove hidden when https://github.com/angular/angular/issues/18310 is resolved
  */
 @Component({
   selector: 'ngx-tab',
   template: `
-    <div [hidden]="!active">
-      <ng-content></ng-content>
-    </div>
+    <div *ngIf="template; then template_content else ng_content"></div>
+    <ng-template #template_content>
+      <div *ngIf="active">
+        <ng-container [ngTemplateOutlet]="template.templateRef"></ng-container>
+      </div>
+    </ng-template>
+    <ng-template #ng_content>
```

Comment:
I don't like the name `#ng_content`, say something like `container` or something like that.

Reply:
I'll change this and the outdated note above.  Are you happy with the `ngxIfTabActive` directive name?

Reply:
Ya thats good